File: presentation_code.ts
// TypeScript
{
    let my_name: string = 'Aaron';
    let my_integer: number = 123;
    let say_hi = (input: number | string): void => console.log(`Hi ${input}`);

    say_hi(my_integer);
    say_hi(my_name);

}

// TypeScript - type error
{
    //let my_name: string = 'Aaron';
    let my_name: string = 123;
    //let my_name: string = undefined;
    function how_many_letters_in_my_name(name: string) {
        let character_count: number = name.length;
        console.log('There are ' + character_count + ' letters in "' + name + '"');
    }
    how_many_letters_in_my_name(my_name);
}

File: presentation_code.R
# R
my_integer <- as.integer(0)
my_name <- "Aaron"

say_hi <- function(input)
    print(paste("Hi", input))

say_hi(my_integer)
say_hi(my_name)

# R - type error
#my_name <- "Aaron"
my_name <- as.integer(123)
#my_name <- NA
how_many_letters_in_my_name <- function(name) {
    character_count <- nchar(name)
    print(paste("There are", character_count, "letters in '", name, "'"))
}
how_many_letters_in_my_name(my_name)